Framework/Core/src/DataRelayer.cxx

```diff
@@ -274,9 +283,13 @@ DataRelayer::RelayChoice
     cachedStateMetrics[cacheIdx] = CacheEntryStatus::PENDING;
     // TODO: make sure that multiple parts can only be added within the same call of
     // DataRelayer::relay
-    PartRef entry{std::move(header), std::move(payload)};
+    PartRef entry{std::move(header), std::move(payloads[0])};
     parts.emplace_back(std::move(entry));
-    assert(header.get() == nullptr && payload.get() == nullptr);
+    auto rest = payloads + 1;
+    for (size_t pi = 0; pi < numPayloads / 2; ++pi) {
```

Comment:
`numPayloads` does not specify the number of messages but message pairs, so no division by 2. Also I would for clarity pass not splitParts - 1, but the total number of payloads, so the upper bound is one less
```suggestion
    for (size_t pi = 0; pi < numPayloads - 1; ++pi) {
```